conans/client/api/conan_api.py

```diff
@@ -98,24 +27,35 @@ def wrapper(api, *args, **kwargs):
             old_curdir = os.getcwd()
         except EnvironmentError:
             old_curdir = None
+
+        if api.quiet:
+            old_stdout = sys.stdout
+            old_stderr = sys.stderr
+            devnull = open(os.devnull, 'w')
```

Comment:
The logger has been removed for simplicity. Anyway, it couldn't mute the output because of the "write" without \n operations (the logger only write new lines)